### PR TITLE
fix: move `c.stream*` to helper

### PR DESCRIPTION
**Pull request by sor4chi**
ref: #1843 

### Author should do the followings, if applicable

- [x] Add tests
- [x] Run tests
- [x] `yarn denoify` to generate files for Deno

### CODE REVIEW

**Comment by yusukebe** [comment]
@sor4chi 

Is this ready to review?

**Comment by sor4chi** [comment]
@yusukebe Yes, please.

File: src/context.ts [hunk 1]
```diff
@@ -375,6 +375,9 @@ export class Context<
     return this.newResponse(null, status)
   }
 
+  /** @deprecated
+   * Use `streamText()` in `hono/helper/streaming` instead of `c.streamText()`. The `c.streamText()` will be removed in v4.
```
Comment (yusukebe) [inline src/context.ts:379]: You can import it from `hono/streaming`. So, `hono/helper/streaming` should be `hono/streaming`.

File: src/context.ts [hunk 2]
```diff
@@ -387,6 +390,9 @@ export class Context<
     return this.stream(cb, arg, headers)
   }
 
+  /** @deprecated
+   * Use `stream()` in `hono/helper/streaming` instead of `c.stream()`. The `c.stream()` will be removed in v4.
```
Comment (yusukebe) [inline src/context.ts:394]: Same as above.

File: src/helper/streaming/index.ts [hunk 3]
```diff
@@ -1,59 +1,21 @@
-import type { Context } from '../../context'
+import type { Context, HeaderRecord } from '../../context'
+import type { StatusCode } from '../../utils/http-status'
 import { StreamingApi } from '../../utils/stream'
 
-interface SSEMessage {
-  data: string
-  event?: string
-  id?: string
+export const stream = (
```
Comment (yusukebe) [inline src/helper/streaming/index.ts:5]: We can make it simpler:

```ts
export const stream = (
  c: Context,
  cb: (stream: StreamingApi) => Promise<void>,
): Response => {
  const { readable, writable } = new TransformStream()
  const stream = new StreamingApi(writable)
  cb(stream).finally(() => stream.close())
  return c.newResponse(readable)
}
```

Users can set the status and the headers with `c.status()` and `c.header(),` so I think it does not need `arg?` and `headers?`. `steamSSE()` also follows this approach.

File: src/helper/streaming/text.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import type { Context, HeaderRecord } from '../../context'
+import { TEXT_PLAIN } from '../../context'
+import type { StatusCode } from '../../utils/http-status'
+import type { StreamingApi } from '../../utils/stream'
+import { stream } from '.'
+
+export const streamText = (
```
Comment (yusukebe) [inline src/helper/streaming/text.ts:7]: Same as above:

```ts
export const streamText = (
  c: Context,
  cb: (stream: StreamingApi) => Promise<void>,
): Response => {
  c.header('content-type', TEXT_PLAIN)
  c.header('x-content-type-options', 'nosniff')
  c.header('transfer-encoding', 'chunked')
  return stream(c, cb)
}
```

**Review by yusukebe — COMMENTED** [review]
@sor4chi 

Thank you for the PR. Take a look the comments!

**Comment by sor4chi** [comment]
Hi, @yusukebe 
Thanks, I fixed.

**Review by yusukebe — APPROVED** [review]
LGTM

**Pull request merged**